### DISCUSSION TITLE
Yolov5 training problem

### DISCUSSION THREAD

**Discussion by dilsadunsal** [ultralytics/yolov5#4656] [Q&A]
Hello, I have trained Yolov5 with my own dataset on Colab. Hyperparameters; image size=640, batch size=16 with 50 epoch. 

Now I would like to train same dataset with different hyperparameter; image size should be 576.

But it is failed. I received memory fault. Whats the reason of this problem?

**Comment by glenn-jocher** [comment]
@siangfei 👋 Hello! Thanks for asking about **CUDA memory issues**. YOLOv5 🚀 can be trained on CPU, [single-GPU](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data), or [multi-GPU](https://github.com/ultralytics/yolov5/issues/475). When training on GPU it is important to keep your batch-size small enough that you do not use all of your GPU memory, otherwise you will see a CUDA Out Of Memory (OOM) Error and your training will crash. You can observe your CUDA memory utilization using either the `nvidia-smi` command or by viewing your console output:

<img width="800" alt="Screenshot 2021-05-28 at 12 19 51" src="https://user-images.githubusercontent.com/26833433/131122318-8141d1ce-13de-44bf-bbdc-487bad442ff0.png">

If you encounter a CUDA OOM error, the steps you can take to reduce your memory usage are:

- Reduce `--batch-size`
- Reduce `--img-size`
- Reduce model size, i.e. from YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s
- Train with [multi-GPU](https://github.com/ultralytics/yolov5/issues/475) at the same `--batch-size`
- Upgrade your hardware to a larger GPU
- Train on free GPU backends with up to 16GB of CUDA memory: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

**Reply by dilsadunsal** [reply]
Thanks for your reply.

 I surprised that I could have trained batch size=16 in my previous training.

You mentioned about multi gpu training. When I use --nproc_per_node... Script, where do I need to determine the epoch number in the command?

**Reply by dilsadunsal** [reply]
> Thanks for your reply.
> 
> I surprised that I could have trained batch size=16 in my previous training.
> 
> You mentioned about multi gpu training. When I use --nproc_per_node... Script, where do I need to determine the epoch number in the command?

![sss](https://user-images.githubusercontent.com/77750296/132053333-fdebbf9a-7c89-47d4-8f68-e7e4eb8da669.PNG)
I also received attached error while trying multi gpu command that you shared the link.
